jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -3186,6 +3186,10 @@ private static MethodHandle dropArgumentsUnsafe(MethodHandle originalHandle, int
 			/*[MSG "K039c", "Invalid parameters"]*/
 			throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K039c")); //$NON-NLS-1$
 		}
+
+		if (valueTypeLength == 0)
+			return originalHandle;
```

Comment:
Would it catch more cases if we move this check into MethodHandle#permuteArguments: https://github.com/eclipse/openj9/blob/cc11a43912e6d9dc0afb0a50e2f9e3848eeeef95/jcl/src/java.base/share/classes/java/lang/invoke/MethodHandle.java#L884

That could code be modified to return itself if the new MT and the MH.type are the same.

Reply:
This needs `{` and `}` for the scope still